ratis-client/src/main/java/org/apache/ratis/client/impl/BlockingImpl.java

```diff
@@ -99,7 +100,7 @@ public RaftClientRequest newRequestImpl() {
           return client.handleReply(request, reply);
         }
       } catch (GroupMismatchException | StateMachineException | TransferLeadershipException |
-          LeaderSteppingDownException e) {
+          LeaderSteppingDownException | AlreadyExistsException e) {
```

Comment:
Should we also avoid retry for `AlreadyClosedException`?

Reply:
Sure.  Just have added it.